stock_resupply_order/models/stock_resupply_order.py

```diff
@@ -0,0 +1,200 @@
+import logging
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class StockResupplyOrder(models.Model):
+    _name = "stock.resupply.order"
+    _inherit = ["mail.thread"]
+    _description = """
+    Creates resupply orders that create procurements
+    following stocks that are already in the destination location
+    """
+
+    location_id = fields.Many2one(
+        "stock.location",
+        required=True,
+        help="The stock location that needs to be restocked",
+    )
+
+    company_id = fields.Many2one(
+        "res.company",
+        "Company",
+        required=True,
+        index=True,
+        default=lambda self: self.env.company,
+    )
+
+    stock_resupply_order_line_ids = fields.One2many(
+        "stock.resupply.order.line",
+        inverse_name="stock_resupply_order_id",
+        help="All product quantities desired at the given location",
+        string="Desired products quantities",
+    )
+
+    procurement_group_id = fields.Many2one(
+        "procurement.group", string="Generated procurement"
+    )
+
+    state = fields.Selection(
+        [
+            ("draft", "Draft"),
+            ("done", "Done"),
+        ],
+        "Status",
+        help=(
+            "Resupply order state: \n"
+            "* draft: The order has not been confirmed yet.\n"
+            "* done: The procurements have been created. \n"
+        ),
+        copy=False,
+        readonly=True,
+        tracking=True,
+        store=True,
+        default="draft",
+    )
+
+    picking_ids = fields.One2many(
+        "stock.picking",
+        compute="_compute_picking_ids",
+        string="Generated stock pickings",
+        readonly=True,
+    )
+
+    picking_count = fields.Integer(
+        string="Picking count", compute="_compute_picking_ids", readonly=True
+    )
+
+    @api.depends("procurement_group_id.stock_move_ids.picking_id")
+    def _compute_picking_ids(self):
+        for record in self:
+            record.picking_ids = record.procurement_group_id.stock_move_ids.picking_id
+            record.picking_count = len(record.picking_ids)
+
+    def action_run(self):
+        if self.procurement_group_id:
+            return self.procurement_group_id
+
+        self.procurement_group_id = self.env["procurement.group"].create(
+            self._get_procurement_group_parameters()
```

Comment:
Just to follow major convention used in the code
```suggestion
            self._prepare_procurement_group_values()
```